test/e2e/suite-ingress/resourcesync.go

```diff
@@ -0,0 +1,216 @@
+		APISIXConfigPath:      "testdata/apisix-gw-config.yaml",
+		IngressAPISIXReplicas: 1,
+		HTTPBinServicePort:    80,
+		APISIXRouteVersion:    "apisix.apache.org/v2beta3",
+	}
+	s := scaffold.NewScaffold(opts)
+	ginkgo.JustBeforeEach(func() {
+		backendSvc, backendPorts := s.DefaultHTTPBackend()
+		ar := fmt.Sprintf(`
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixRoute
+metadata:
+ name: httpbin-route
+spec:
+ http:
+ - name: rule1
+   match:
+     hosts:
+     - httpbin.org
+     paths:
+       - /ip
+   backends:
+   - serviceName: %s
+     servicePort: %d
+`, backendSvc, backendPorts[0])
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ar))
+		err := s.EnsureNumApisixUpstreamsCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of upstreams")
+		err = s.EnsureNumApisixRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+
+		ing := fmt.Sprintf(`
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+    kubernetes.io/ingress.class: apisix
+  name: ingress-extensions-v1beta1
+spec:
+  rules:
+  - host: local.httpbin.org
+    http:
+      paths:
+      - path: /headers
+        pathType: Exact
+        backend:
+          serviceName: %s
+          servicePort: %d
+`, backendSvc, backendPorts[0])
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ing))
+
+		err = s.ApisixConsumerKeyAuthCreated("foo", "foo-key")
+		assert.Nil(ginkgo.GinkgoT(), err)
+	})
+
+	ginkgo.It("for modified resource sync consistency", func() {
+		// crd resource sync interval
+		readyTime := time.Now().Add(60 * time.Second)
+
+		routes, _ := s.ListApisixRoutes()
+		assert.Len(ginkgo.GinkgoT(), routes, 2)
+
+		consumers, _ := s.ListApisixConsumers()
+		assert.Len(ginkgo.GinkgoT(), consumers, 1)
+
+		for _, route := range routes {
+			_ = s.CreateApisixRouteByApisixAdmin(id.GenID(route.Name), []byte(`
+{
+	"methods": ["GET"],
+	"uri": "/anything",
+	"plugins": {
+		"key-auth": {}
+	},
+	"upstream": {
+		"type": "roundrobin",
+		"nodes": {
+			"httpbin.org": 1
+		}
+	}
+}`))
+		}
+
+		for _, consumer := range consumers {
+			_ = s.CreateApisixConsumerByApisixAdmin([]byte(fmt.Sprintf(`
+{
+	"username": "%s",
+	"plugins": {
+		"key-auth": {
+			"key": "auth-one"
+		}
+	}
+}`, consumer.Username)))
+		}
+
+		_ = s.NewAPISIXClient().
+			GET("/ip").
+			WithHeader("Host", "httpbin.org").
+			Expect().
+			Status(http.StatusNotFound)
+
+		_ = s.NewAPISIXClient().
+			GET("/headers").
+			WithHeader("Host", "local.httpbin.org").
+			Expect().
+			Status(http.StatusNotFound)
+
+		waitTime := time.Until(readyTime).Seconds()
+		time.Sleep(time.Duration(waitTime) * time.Second)
+
+		_ = s.NewAPISIXClient().
+			GET("/ip").
+			WithHeader("Host", "httpbin.org").
+			Expect().
+			Status(http.StatusOK)
+
+		_ = s.NewAPISIXClient().
+			GET("/headers").
+			WithHeader("Host", "local.httpbin.org").
+			Expect().
+			Status(http.StatusOK)
+
+		consumers, _ = s.ListApisixConsumers()
```

Comment:
I think we should bind key-auth authentication for a route and then verify, what do you think?

Reply:
Sure!